source/features/mark-merge-commits-in-list.tsx

```diff
@@ -0,0 +1,56 @@
+import './mark-merge-commits-in-list.css';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getRepoGQL} from '../libs/utils';
+
+interface CommitInfo {
+	parents: {
+		totalCount: number | string;
+	};
+}
+
+const getCommitParentCount = async (commits: string[]): Promise<Record<string, CommitInfo>> => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			${commits.map((commit: string) => `
+				${api.escapeKey(commit)}: object(expression: "${commit}") {
+					... on Commit {
+							parents{
+								totalCount
+							}
+						}
+					}
+			`).join('\n')}
+		}
+	`);
+
+	return repository;
```

Comment:
Parse the API response and just return the data you want: a list of commit hashes that are merge commits. This way you won’t have to type CommitInfo. Also the function name would become `filterMergeCommits`